From 30827b15cc232a75b510a6e54efae579de7f5bf4 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Wed, 5 Feb 2020 15:04:07 +0100
Subject: [PATCH] PONRTSYS-6642: Activate SERIAL_EARLYCON if we have a serial
 driver

The option CONFIG_SERIAL_EARLYCON  should only be activated when we also
have the lantiq serial driver in the image otherwise we get this link
error:

mips-openwrt-linux-musl-ld: drivers/built-in.o: in function `setup_earlycon':
(.init.text+0x24f4): undefined reference to `uart_parse_earlycon'
Makefile:1016: recipe for target 'vmlinux' failed
---
 arch/mips/lantiq/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/lantiq/Kconfig b/arch/mips/lantiq/Kconfig
index bc7ca091db41..b5cbd4e9bd05 100755
--- a/arch/mips/lantiq/Kconfig
+++ b/arch/mips/lantiq/Kconfig
@@ -48,7 +48,7 @@ config SOC_GRX500
 	select SYS_SUPPORTS_MIPS_CPS
 	select SYS_SUPPORTS_MIPS_CMP
 	select SYS_SUPPORTS_ZBOOT
-	select SERIAL_EARLYCON
+	select SERIAL_EARLYCON if SERIAL_LANTIQ
 	select HW_HAS_PCI
 	select ARCH_SUPPORTS_MSI
 	select ARCH_HAS_RESET_CONTROLLER
